src/results.h

```diff
@@ -8,23 +8,20 @@ using namespace arma;
 struct Results {
   mat beta;
   uword passes;
-  std::vector<double> primals;
-  std::vector<double> duals;
+  std::vector<std::vector<double>> diagnosticsLoss;
```

Comment:
just nitpicking here, but variable names should be lower snake case

```suggestion
  std::vector<std::vector<double>> diagnostics_loss;
```